Gem/Code/Source/MultiplayerSampleSystemComponent.cpp

```diff
@@ -108,32 +109,33 @@ namespace MultiplayerSample
         return AZ::TICK_PLACEMENT + 2;
     }
 
-    Multiplayer::NetworkEntityHandle MultiplayerSampleSystemComponent::OnPlayerJoin(
-        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const Multiplayer::MultiplayerAgentDatum& agentDatum)
+    void MultiplayerSampleSystemComponent::OnPlayerJoin(
+        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const Multiplayer::MultiplayerAgentDatum& agentDatum, AzFramework::EntitySpawnCallback playerSpawnedCallback)
     {
-        AZStd::pair<Multiplayer::PrefabEntityId, AZ::Transform> entityParams = AZ::Interface<IPlayerSpawner>::Get()->GetNextPlayerSpawn();
+        AZStd::pair<AZ::Data::Asset<AzFramework::Spawnable>, AZ::Transform> entityParams = AZ::Interface<IPlayerSpawner>::Get()->GetNextPlayerSpawn();
 
-        Multiplayer::INetworkEntityManager::EntityList entityList =
-            AZ::Interface<Multiplayer::IMultiplayer>::Get()->GetNetworkEntityManager()->CreateEntitiesImmediate(
-            entityParams.first, Multiplayer::NetEntityRole::Authority, entityParams.second, Multiplayer::AutoActivate::DoNotActivate);
-
-        for (Multiplayer::NetworkEntityHandle subEntity : entityList)
+        // Callback to move the player to the desired position before it's spawned into the world
+        auto preSpawnCallback = [entityParams](
+            AzFramework::EntitySpawnTicket::Id /*ticketId*/,
+            AzFramework::SpawnableEntityContainerView entities
+            )
         {
-            subEntity.Activate();
-        }
+            const AZ::Entity* e = *entities.begin();
+            if (AZ::TransformInterface* transform = e->GetTransform())
+            {
+                transform->SetWorldTM(entityParams.second);
+            }
+        };
 
-        Multiplayer::NetworkEntityHandle controlledEntity;
-        if (!entityList.empty())
+        auto ticket = AZStd::make_unique<AzFramework::EntitySpawnTicket>(entityParams.first);
+        if (ticket->IsValid())
```

Comment:
Does this need a log message or similar if the ticket is not valid? Is it ok to silently not spawn?